recipe/meta.yaml

```
@@ -1,45 +1,56 @@
 {% set name = "cuda-python" %}
-{% set version = "11.8.1" %}
+{% set version = "12.0.0" %}
 {% set major_version = version.split(".")[0] %}
 {% set minor_version = version.split(".")[1] %}
 {% set patch_version = version.split(".")[2] %}
+{% set target_name = "x86_64-linux" %}  # [linux64]
+{% set target_name = "ppc64le-linux" %}  # [ppc64le]
+{% set target_name = "sbsa-linux" %}  # [aarch64]
+{% set target_name = "x64" %}  # [win]
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/NVIDIA/cuda-python/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: ef54ada5d607387f59950cfb3c3852e5816b62824402d289227b8066e8353489
+  sha256: 7df0f84c4b6210112353374a7d992bfd5bba9c97358b39ea03b61bd634ce9c8a
 
 build:
-  number: 2
-  skip: true  # [py<37 or osx or cuda_compiler_version != "11.8"]
+  number: 0
+  skip: true  # [py<37 or osx or cuda_compiler_version != "12.0"]
   script:
     - {{ PYTHON }} -m pip install . --no-deps -vv
+  script_env:
+    # Ensure that CUDA includes can be found by the host compiler
+    - export CUDA_HOME="${BUILD_PREFIX}/targets/{{ target_name }}"
   run_exports:
-    - cuda-python >=11.7.1,<12  # we need this because CUDA Python 11.7.1 has ABI breaks, see conda-forge/cuda-python-feedstock#15
-  ignore_run_exports:
-    - cudatoolkit
+    - cuda-python >=12.0.0
+  # TODO: Do we need to ignore anything in CUDA 12?
+  #ignore_run_exports:
+  #  - cuda-cudart
+  #  - cuda-nvrtc
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    - cuda-nvrtc-dev
+    - cuda-profiler-api
```

Guessing these are added to `build` to workaround the fact that `cuda-python` doesn't support splayed layouts? Seems reasonable.

Maybe we can add a comment? Perhaps something like this (though please feel free to modify it)

```suggestion
    # Workaround absence of splayed layout support
    - cuda-nvrtc-dev
    - cuda-profiler-api
```

---

```suggestion
    # The packages with CUDA headers need to be added to "build" so that
    # cuda-python can find and parse the headers. The parsing code in
    # cuda-python doesn't support splayed layouts, i.e. packages split between
    # "build" and "host", because $CUDA_HOME can only point to build or host.
    - cuda-nvrtc-dev
    - cuda-profiler-api
```

---

xref: https://github.com/NVIDIA/cuda-python/issues/46